Find a functionally equivalent code:android.app.Notification.safeCharSequence:COMMENT
Method Modifier: public      static      hidden      
Comment:/**
 * Make sure this CharSequence is safe to put into a bundle, which basically
 * means it had better not be some custom Parcelable implementation.
 * @hide
 */

@@ -1,4 +1,9 @@
 {
+    if (cs == null)
+        return cs;
+    if (cs.length() > MAX_CHARSEQUENCE_LENGTH) {
+        cs = cs.subSequence(0, MAX_CHARSEQUENCE_LENGTH);
+    }
     if (cs instanceof Parcelable) {
         Log.e(TAG, "warning: " + cs.getClass().getCanonicalName() + " instance is a custom Parcelable and not allowed in Notification");
         return cs.toString();

